azurerm/data_source_arm_scheduled_time_test.go

```diff
@@ -0,0 +1,444 @@
+					resource.TestCheckResourceAttr(dataSourceName, "next_run_time", formattedExpectedTime),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAzureRMScheduledTime_Monthly_Tomorrow(t *testing.T) {
+	dataSourceName := "data.azurerm_scheduled_time.test"
+
+	now := time.Now().UTC()
+	scheduletime := now.Add(time.Duration(24) * time.Hour) //1 day
+	dayofmonth := scheduletime.Day()
+
+	config := testAccDataSourceAzureRMScheduledTime_Monthly(dayofmonth, scheduletime)
+
+	expectedTime := time.Date(scheduletime.Year(), scheduletime.Month(), scheduletime.Day(), scheduletime.Hour(), 0, 0, 0, time.UTC)
+	formattedExpectedTime := expectedTime.Format(time.RFC3339)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "day_of_month", strconv.Itoa(dayofmonth)),
+					resource.TestCheckResourceAttr(dataSourceName, "hour", strconv.Itoa(now.Hour())),
+					resource.TestCheckResourceAttr(dataSourceName, "minute", "0"),
+					resource.TestCheckResourceAttr(dataSourceName, "second", "0"),
+					resource.TestCheckResourceAttr(dataSourceName, "frequency", "Month"),
+					resource.TestCheckResourceAttr(dataSourceName, "next_run_time", formattedExpectedTime),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAzureRMScheduledTime_Monthly_Next_Month(t *testing.T) {
+	dataSourceName := "data.azurerm_scheduled_time.test"
+
+	now := time.Now().UTC()
+	scheduletime := now.Add(25 * time.Duration(24) * time.Hour) //25 days
+	dayofmonth := scheduletime.Day()
+
+	config := testAccDataSourceAzureRMScheduledTime_Monthly(dayofmonth, scheduletime)
+
+	expectedTime := time.Date(scheduletime.Year(), scheduletime.Month(), scheduletime.Day(), scheduletime.Hour(), 0, 0, 0, time.UTC)
+	formattedExpectedTime := expectedTime.Format(time.RFC3339)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "day_of_month", strconv.Itoa(dayofmonth)),
+					resource.TestCheckResourceAttr(dataSourceName, "hour", strconv.Itoa(now.Hour())),
+					resource.TestCheckResourceAttr(dataSourceName, "minute", "0"),
+					resource.TestCheckResourceAttr(dataSourceName, "second", "0"),
+					resource.TestCheckResourceAttr(dataSourceName, "frequency", "Month"),
+					resource.TestCheckResourceAttr(dataSourceName, "next_run_time", formattedExpectedTime),
+				),
+			},
+		},
+	})
+}
+
+func TestAccDataSourceAzureRMScheduledTime_Monthly_Last_Month(t *testing.T) {
+	dataSourceName := "data.azurerm_scheduled_time.test"
+
+	now := time.Now().UTC()
+	scheduletime := now.Add(-25 * time.Duration(24) * time.Hour) //25 days
```

Comment:
I think it'd be better to construct a date as below - specifying `now.Month() - 1`?

Reply:
Yeah, that was my first attempt but it won't work, if i am right. Can you validate it pls?
If its the end of the month the test will fail. Eg. at the end of march the `now` is `2017-03-31` if the scheduled date is constructed with `month() - 1` it will become `2017-02-31` which is in golang the same as `2017-03-03` so the assertion will fail because we expect March but will get April. 

I don't have any idea to workaround this so right now i removed this test.